### PR TITLE
Updating Github Templates

### PR DESCRIPTION
**Pull request by malachi-constant**
***Issue***
- Time to revamp github templates and make use of Issue Template Form support (Beta)

***Description of changes:***
- Updates to pull request template to add more detail 
- Updates to bug report template, utilizes forms to enable required values and adds more sections
- Adds discussion link to Public Data Wrangler Slack channel

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: be50ada1ba21848c46538cadcb3ab00b2f04db69
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b753aecb-e9b5-4fbf-b8be-8a97fea23876%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: c26ca2f0b2bfde3173862361bf9b71f2e2e3e914
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=33304228-4e66-4cb3-b09a-823ac2798650%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -13,4 +13,10 @@ A clear and concise description of what the problem is. Ex. I'm always frustrate
...
 A clear and concise description of what you want to happen.
 
-*P.S. Don't attach files. Please, prefer add code snippets directly in the message body.*
+**Describe alternatives you've considered**
+A clear and concise description of any alternative solutions or features you've considered.
+
+**Additional context**
+Add any other context or screenshots about the feature request here.
+
+*P.S. Don't attach files. Please, prefer add code snippets directly in the message body.*
```
Comment (jaidisido) [inline .github/ISSUE_TEMPLATE/feature_request.md:22]: Nitpick - the original message had bad grammar (not yours :) ), could we change it to:
`*P.S. Please Don't attach files. Add code snippets directly in the message body instead.*`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+<!-- please choose -->
+- Feature
+- Bugfix
+- Refactoring
 
+### Purpose
+- <long purpose of this pull request>
+- <Environment if this PR depends on>
 
-By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
```
Comment (jaidisido) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: This is a required sentence at the end of AWS PRs I am afraid, could we put it back please?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
-*Description of changes:*
+### Feature or Bugfix
+<!-- please choose -->
+- Feature
+- Bugfix
+- Refactoring
 
+### Purpose
+- <long purpose of this pull request>
+- <Environment if this PR depends on>
```
Comment (jaidisido) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: Not sure I understand this one?

**Review by jaidisido — COMMENTED** [review]
Looking good, left minor comments

**Review by kukushking — APPROVED** [review]

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: c5d0bb5879882c2422f81fa43261d6a865a2d66d
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a3addd64-849b-48cd-b325-edcbec80919d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 0727f2e9c445db9fdcde37a17413ddd1c45ff5b9
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c4ba25f5-e6f2-4f82-8aef-8b394ccf739d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**